app/scripts/controllers/new_proposal.js

```diff
@@ -8,52 +8,36 @@ angular.module('admissionSystemApp')
-      });
-      SpecofferDictionaryService.getSpecoffersTypes().then(function (SpecoffersTypes) {
-        $scope.specofferTypesOptions = SpecoffersTypes;
-      });
-      SpecofferDictionaryService.getEduformTypes().then(function (eduformtypes) {
-         $scope.eduFormTypesOptions = eduformtypes;
-      });
-
-      SpecofferDictionaryService.getTimePeriodCourseIds().then(function (timePeriodCourseIds) {
-        $scope.timePeriodCourseId = timePeriodCourseIds;
-      });
-
-  $scope.brosweOrEditSpecoffer = function  (specofferId) {
-    SpecoffersService.getEntireSpecoffer(specofferId).then(function(res){
-      _.merge($scope.entireSpecoffer.subjects, res.subjects);
-      _.merge($scope.entireSpecoffer.benefits, res.benefits);
-      _.merge($scope.entireSpecoffer.specoffer, res.specoffer);
+    $q.all([
+      SpecofferDictionaryService.getAllDepartments(),
+      SpecofferDictionaryService.getSpecoffersTypes(),
+      SpecofferDictionaryService.getEduformTypes(),
+      SpecofferDictionaryService.getTimePeriodCourseIds()
+    ])
+    .then(function(promisesResult) {
+      $scope.departmentId = promisesResult[0];
+      $scope.specofferTypesOptions = promisesResult[1];
+      $scope.eduFormTypesOptions = promisesResult[2];
+      $scope.timePeriodCourseId = promisesResult[3];
     });
-  };
-  // $scope.brosweOrEditSpecoffer(407);
-
-      // DEMO FOR GEtting ALL Departments
-      // SpecofferDictionaryService.getAllDepartments().then(function (allDepartments) {
-        // console.log('allDepartments',allDepartments);
-      // })
 
-      // DEMO FOR GEtting ALL specialities
-      // SpecofferDictionaryService.getAllSpecialties().then(function (specialties) {
-        // console.log('Specialties',specialties);
-      // })
-
-  $scope.sendToServer = function (entireSpecoffer) {
-    $scope.entireSpecoffer.specoffer.note = 'some note';
-    SpecoffersService.addOrEditSpecoffer(entireSpecoffer);
-  };
+    $scope.brosweOrEditSpecoffer = function  (specofferId) {
+      SpecoffersService.getEntireSpecoffer(specofferId).then(function(res){
+        _.merge($scope.entireSpecoffer.subjects, res.subjects);
+        _.merge($scope.entireSpecoffer.benefits, res.benefits);
+        _.merge($scope.entireSpecoffer.specoffer, res.specoffer);
+      });
+    };
+    $scope.brosweOrEditSpecoffer(384);
 
-      // delete this
-      // $scope.$watch('entireSpecofferCopy', function (newVal) {
-      //   console.log('entireSpecofferCopy watch', newVal);
-      // }, true);
+    $scope.sendToServer = function (entireSpecoffer) {
+      $scope.entireSpecoffer.specoffer.note = 'some note';
+      SpecoffersService.addOrEditSpecoffer(entireSpecoffer);
+    };
 
-      // $scope.$watch('entireSpecoffer', function (newVal) {
-      //   console.log('entireSpecoffer watch', newVal);
-      // }, true);
+    $scope.delete = function (entireSpecoffer) {
+      SpecoffersService.deleteEntireSpecoffer(entireSpecoffer);
```

Comment:
Why do we need to send entire specoffer to delete? we just need to sent its id.

May it happen that user changed something (deleted benefits) and sending this new data? Will it be deleted? As I can see in line 148 of specofferService, it will not be deleted.